src/Internal/Types/ScriptLookups.purs

```diff
@@ -995,13 +1012,27 @@ processConstraint mpsMap osMap = do
     MustReferenceOutput refInput -> runExceptT do
       _cpsToTxBody <<< _referenceInputs %= Set.insert refInput
     MustMintValue mpsHash red tn i scriptRefUnspentOut -> runExceptT do
-      case scriptRefUnspentOut of
+      isNative <- case scriptRefUnspentOut of
         Nothing -> do
-          plutusScript <-
-            except $ unwrap <$> lookupMintingPolicy mpsHash mpsMap
-          ExceptT $ attachToCps attachPlutusScript plutusScript
-        Just scriptRefUnspentOut' ->
-          ExceptT $ processScriptRefUnspentOut mpsHash scriptRefUnspentOut'
+          mp <- except $ lookupMintingPolicy mpsHash mpsMap
```

Comment:
Please remove the ability to mint with native script using `MustMintValue` constraint. The users should call `MustMintValueUsingNativeScript` only (having two ways of doing things is bad)